BGL/include/CGAL/boost/graph/shortest_path.h

```diff
@@ -0,0 +1,160 @@
+  {
+    using vertex_descriptor = typename boost::graph_traits<Graph>::vertex_descriptor;
+    using edge_descriptor = typename boost::graph_traits<Graph>::edge_descriptor;
+
+  public:
+    vertex_descriptor destination_vd;
+
+    void initialize_vertex(const vertex_descriptor& /*s*/, const Graph& /*g*/) const {}
+    void examine_vertex(const vertex_descriptor& /*s*/, const Graph& /*g*/) const {}
+    void examine_edge(const edge_descriptor& /*e*/, const Graph& /*g*/) const {}
+    void edge_relaxed(const edge_descriptor& /*e*/, const Graph& /*g*/) const {}
+    void discover_vertex(const vertex_descriptor& /*s*/, const Graph& /*g*/) const {}
+    void edge_not_relaxed(const edge_descriptor& /*e*/, const Graph& /*g*/) const {}
+    void finish_vertex(const vertex_descriptor& vd, const Graph& /* g*/) const
+    {
+      if (vd == destination_vd)
+        throw Dijkstra_end_exception();
+    }
+  };
+} // namespace internal
+
+/*!
+* \ingroup PkgBGLTraversal
+* Computes the shortest path between two vertices in a graph `g`
+*
+*@tparam Graph a model of the concept `HalfedgeListGraph`
+* @param vs source vertex
+* @param vt target vertex
+* @param g the graph
+* @param halfedge_sequence_oit the output iterator holding the output sequence
+* of halfedges that form the shortest path from `vs` to `vt` on `g`
+* @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+* ** edge_weight_map : @todo deal with input mesh with no internal pmap.
+*                       default in boost is `get(boost::edge_weight, mesh)`
+*
+*/
+template<typename Graph,
+         typename OutputIterator,
+         typename NamedParameters = parameters::Default_named_parameters>
+OutputIterator shortest_path_between_two_vertices(
+  const typename boost::graph_traits<Graph>::vertex_descriptor vs,//source
+  const typename boost::graph_traits<Graph>::vertex_descriptor vt,//target
+  const Graph& g,
+  OutputIterator halfedge_sequence_oit,
+  const NamedParameters& np = parameters::default_values())
+{
+  using vertex_descriptor = typename boost::graph_traits<Graph>::vertex_descriptor;
+  using halfedge_descriptor = typename boost::graph_traits<Graph>::halfedge_descriptor;
+
+  using Pred_umap = std::unordered_map<vertex_descriptor, vertex_descriptor>;
+  using Pred_pmap = boost::associative_property_map<Pred_umap>;
+
+  using parameters::get_parameter;
+  using parameters::choose_parameter;
+
+  const auto w_map = choose_parameter(get_parameter(np, internal_np::edge_weight),
+                                      get(boost::edge_weight, g));
+
+  Pred_umap predecessor;
+  Pred_pmap pred_pmap(predecessor);
+
+  internal::Stop_at_target_Dijkstra_visitor<Graph> vis;
+  vis.destination_vd = vt;
+  try
+  {
+    boost::dijkstra_shortest_paths(g, vs,
+      boost::predecessor_map(pred_pmap)
+      .visitor(vis)
+      .weight_map(w_map));
+  }
+  catch (const std::exception& e){}
```

Comment:
That looks dangerous to catch and dismiss all `std::exception` exceptions. Imagine the supplied "edge weight" has a bug in its implementation, and an assertion throws a `CGAL::Failure_exception`.  It would be silently dismissed.

Let's catch only the exception the normal behavior of the code is supposed to throw. And remove the name `e` of the unused variable to avoid a warning.

```suggestion
  catch (const internal::Dijkstra_end_exception&){}
```